### ISSUE TITLE
Jakarta constraint `@PositiveOrZero` overwrites `@Digits` resulting in invalid objects

### ISSUE THREAD

**Issue by philipp-mueller-exxeta** [instancio/instancio#1150]
**Describe the bug**
After generating a bean with the `@Given` annotation with bean validation enabled, some constraints will be ignored depending on the order in which the annotations are written.

This one respects the constraints:

```java
@PositiveOrZero
@Digits(integer = 2, fraction = 5)
private BigDecimal goodExample;
```

This one will not:

```java
@Digits(integer = 2, fraction = 5)
@PositiveOrZero
private BigDecimal badExample;
```

**Expected result**
Generate two BigDecimals with two integer digits and five fraction digits

**Actual result**
The Integer digits of the second Example will exceed the limit of two. Fraction digits are generated correctly.

**Additional information**
The order of annotations should be irrelevant for validation

**Link to a minimal sample project reproducing the issue or runnable sample code**
[sample project](https://github.com/philipp-mueller-exxeta/instancio-bug-example)
Run the InstancioValidationBugTest to see the issue

- Java version: 21
- Instancio version: Instancio-junit-5.0.1

**Comment by armandino** [comment]
@philipp-mueller-exxeta thanks for raising this issue and the sample project! I agree that _ideally_ it should behave consistently, regardless of the order in which these annotations are declared. I'll look into it once I have some time.

**Comment by armandino** [comment]
Hi @philipp-mueller-exxeta, sorry about the delay. After looking into it, I don' t see a simple solution to this issue. The problem is that there are a lot of possible combinations of annotations such as `Digits`, `Min`, `Max`, `Positive`, `Negative`, etc. It's tricky to handle all possible combinations of these annotations to generate valid values. This will require some extensive (potentially breaking) changes.

I'll keep this issue open in case somebody wants to take this on and submit a PR.

**Comment by armandino** [comment]
@philipp-mueller-exxeta I just released version `5.2.1` containing a fix for this bug. Please let me know if you run into any other issues.
